### PR TITLE
Update MP Config version 3.0

### PR DESCRIPTION
**Pull request by jansupol**
Signed-off-by: jansupol <jan.supol@oracle.com>

### CODE REVIEW

**Comment by jansupol** [comment]
Added [CQ for Helidon Config 3.0.0-M1](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23994)
Added [CQ for Smallrye 3.0.0-RC1](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23996)

**Review by senivam — APPROVED** [review]
LGTM, can be merged after CQs' approval.

**Pull request merged**
